gorb/gorb.go

```diff
@@ -0,0 +1,323 @@
+type Pulse struct {
+	TypeHealthcheck string          `json:"type"`
+	Args            HealthcheckArgs `json:"args"`
+	Interval        string          `json:"interval"`
+}
+
+// BackendConfig defines the backend configuration
+type BackendConfig struct {
+	Host   string `json:"host"`
+	Port   int    `json:"port"`
+	Method string `json:"method"`
+	Weight int    `json:"weight"`
+	Pulse  Pulse  `json:"pulse"`
+}
+
+// Backend defines the backend configuration + service name
+type backend struct {
+	ServiceName   string
+	BackendConfig BackendConfig
+}
+
+type loopbackAction string
+
+const (
+	addLoopback    loopbackAction = "add"
+	deleteLoopback loopbackAction = "del"
+)
+
+// New creates a gorb handler
+func New(serverBaseURL string, instanceIP string, drainDelay time.Duration, servicesDefinition string, backendWeight int, backendMethod string, vipLoadbalancer string, manageLoopback bool, backendHealthcheckInterval string, interfaceProcFsPath string) (controller.Updater, error) {
+	if serverBaseURL == "" {
+		return nil, errors.New("unable to create Gorb updater: missing server ip address")
+	}
+	initMetrics()
+	log.Infof("Gorb server url: %s, drainDelay: %v, instance ip adddress: %s, vipLoadbalancer: %s ", serverBaseURL, drainDelay, instanceIP, vipLoadbalancer)
+
+	backendDefinitions := []backend{}
+	servicesDefinitionArr := strings.Split(servicesDefinition, ",")
+
+	var backendDefinition backend
+	for _, service := range servicesDefinitionArr {
+
+		servicesArr := strings.Split(service, ":")
+		port, err := strconv.Atoi(servicesArr[1])
+		if err != nil {
+			return nil, errors.New("Unable to convert port form string to int")
+		}
+
+		args := HealthcheckArgs{
+			Method: "GET",
+			Path:   "/",
+			Expect: "404",
+		}
+		pulse := Pulse{
+			Args:            args,
+			TypeHealthcheck: "http",
+			Interval:        backendHealthcheckInterval,
+		}
+		backendDefinition = backend{
+			ServiceName: servicesArr[0],
+			BackendConfig: BackendConfig{
+				Host:   instanceIP,
+				Port:   port,
+				Method: backendMethod,
+				Weight: backendWeight,
+				Pulse:  pulse,
+			},
+		}
+
+		backendDefinitions = append(backendDefinitions, backendDefinition)
+	}
+
+	var httpClient = &http.Client{
+		Timeout: time.Second * 5,
+	}
+
+	return &gorb{
+		serverBaseURL:       serverBaseURL,
+		drainDelay:          drainDelay,
+		instanceIP:          instanceIP,
+		vipLoadbalancer:     vipLoadbalancer,
+		backend:             backendDefinitions,
+		httpClient:          httpClient,
+		manageLoopback:      manageLoopback,
+		interfaceProcFsPath: interfaceProcFsPath,
+	}, nil
+}
+
+type gorb struct {
+	serverBaseURL       string
+	drainDelay          time.Duration
+	instanceIP          string
+	vipLoadbalancer     string
+	httpClient          *http.Client
+	backend             []backend
+	manageLoopback      bool
+	interfaceProcFsPath string
+}
+
+func (g *gorb) Start() error {
+	return nil
+}
+
+func (g *gorb) manageLoopBack(action loopbackAction) error {
```

Comment:
Can we move this function down to below/above `Update` so we don't have to jump around the file so much when following the control flow.

Reply:
lol - use a proper IDE :-P

Reply:
It's just good practice, place things you use close to where you use them. Same reason we don't declare all our variables at the top of a file anymore :).